programs/disks/disks.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <Common/Config/ConfigProcessor.h>
+#include <Poco/Util/Application.h>
+#include <functional>
+#include "Path.h"
+
+namespace DB
+{
+
+class IDisk;
+using DiskPtr = std::shared_ptr<IDisk>;
+using DisksMap = std::map<String, DiskPtr>;
+
+class Context;
+
+using ProgramOptionsDescription = boost::program_options::options_description;
+using CommandLineOptions = boost::program_options::variables_map;
+
+class DisksApp;
+
+struct CommandDescription
+{
+    String command_name;
+    std::optional<ProgramOptionsDescription> command_option_description;
+    String description;
+    String usage;
+    void (DisksApp::*arg_func)(void);
```

Comment:
Let's get rid of `void (DisksApp::*arg_func)(void);`.
You can just make a class for each command, with inheritance from `ICommand` and using `command_descriptions = std::unordered_map<std:string, CommandPtr>`, where `CommandPtr=std::shared_ptr<ICommand>`. Each will override virtual methods `execute()`, `validateArguements(...)`, `getHelpMessage()`, ...